client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/DateTimeParsingUtils.java

```diff
@@ -212,4 +214,86 @@ public static double getDistanceToISOLocalDate(CharSequence input) {
         return base + ((1d - base) / (distance + 1));
     }
 
+    /**
+     * returns a value that represents how close is the value to the format YYYY-MM-DD
+     *
+     * @param input
+     * @return
+     */
+    public static double getDistanceToDateTime(CharSequence input) {
+
+        if (input == null) {
+            return H_REACHED_BUT_NULL;
+        }
+
+        try {
+            new SimpleDateFormat("YYYY-MM-DD HH:MM").parse(input.toString());
+            /*
+                due to the simplification later on, still must make
+                sure to get a 1 if no exception is thrown
+             */
+            return 1d;
```

Comment:
if this is a "distance", I guess that we should rather return a 0? 
we used 1 when dealing with satisfied heuristic (ie covered target), not for distances

Reply:
you're right. I did not have a test for that case.

Reply:
There was a comment in the code that states the following:        
//recall h in [0,1] where the highest the distance the closer to 0
So, this menas that (according to this comment)
- h should be 1 when the value is successfully parsed
- h should be 0 when the distance to successfully parsing the input

Reply:
I think I understand your point now. Distance should be 0 closer to your target, >0 further away from your target. But the "h" does not mean "distance", but rather "heuristic". So, I will rename all method names in this class to explicitly say that we are returning a "heuristic" value in [0,1], not a "distance".